tools/testing/pytest/tests/test_dash_format_converter.py

```diff
@@ -0,0 +1,115 @@
+# *******************************************************************************
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+
+import os
+import subprocess
+import pytest
+from pathlib import Path
+
+
+@pytest.fixture
+def temp_files(tmp_path):
+    """
+    Creates temporary input and output files for the test and returns their paths.
+    """
+    input_file = tmp_path / "input_file.txt"
+    output_file = tmp_path / "formatted.txt"
+    return input_file, output_file
+
+
+@pytest.fixture(scope="session")
+def converter_path():
+    """
+    Determines the correct way to call dash_format_converter.
+
+    - If running inside Bazel, use TEST_SRCDIR to find the `py_binary` target.
+    - If running manually, call the Python script directly.
+    """
+    if "TEST_SRCDIR" in os.environ:
```

Comment:
Doesn't matter much, but somehow I feel it would be nice if all our tools follow the same approach.
Whether we want to test via subprocess or via call to main().